src/content/tutorials/en/intro-to-p5-strands.mdx

```diff
@@ -0,0 +1,744 @@
+- [`baseNormalShader`](/reference/p5/baseNormalShader): Applied by calling `normalMaterial()`.
+- [`baseStrokeShader`](/reference/p5/baseStrokeShader): Applied unless `noStroke()` is called, draws stroke on objects.
+- [`baseFilterShader`](/reference/p5/baseFilterShader): A template for making post-processing filters such as those provided by p5.js, like `filter(BLUR)`.
+
+Now that we've built our first p5.strands modification, let's create a more complex scene! We'll use `baseColorShader` and `baseStrokeShader` to create 3D objects, then apply post-processing with `baseFilterShader` to enhance the final result.
+
+## Building a scene
+### Instancing particles
+Because the GPU excels at parallel computation, it can draw thousands or millions of particles simultaneously. We can do this with a technique called **GPU Instancing**.
+
+In GPU instancing, we ask the GPU to draw multiple copies of the same object, each with a unique ID (from 0 to n-1). We can then position each instance based on its ID. For example, placing objects at coordinates `[ID, 0, 0]` would create a line along the x-axis.
+
+In p5.js, instancing is available via optional parameters for `endShape` and `model`. It does also require a custom shader to work. In our case, let's use `model` and build a sphere shape.
+
+```js
+particleModel = buildGeometry(() => sphere(10, 2, 4));
+```
+
+Let's start by offsetting instances along the x-axis using `baseColorShader()`:
+
+<EditableSketch code={`
+let instancingShader;
+let instancingStrokeShader;
+let particleModel;
+
+function instancingCallback() {
+	getWorldInputs((inputs) => {
+		// Offset each instance by its ID
+		inputs.position.x += instanceID() * 20;
+    // Move the particles left on the screen
+    inputs.position.x -= 150;
+		return inputs;
+	});
+}
+
+async function setup() {
+	createCanvas(300, 300, WEBGL);
+  particleModel = buildGeometry(() => sphere(10, 10, 2));
+	instancingShader = baseColorShader().modify(instancingCallback);
+  instancingStrokeShader = baseStrokeShader().modify(instancingCallback);
+}
+
+function draw() {
+	background(0);
+	orbitControl();
+	shader(instancingShader)
+  strokeShader(instancingStrokeShader)
+	model(particleModel, 16);
+}
+`} />
+
+The getWorldInputs() callback receives an inputs struct containing data about the current vertex: `position`, `normal`, `texCoord`, and `color`. The "world" part indicates that this runs after JavaScript transformations like translate() or scale() have been applied.
+
+<Callout title='World Space vs. Object Space'>
+Moving in world space is like moving relative to the entire scene, while object transformations are relative to the object's center.
+</ Callout>
+
+Now let's distribute our particles in a more interesting pattern - placing them randomly on a sphere:
+
+<EditableSketch code={`
+let instancingShader;
+let instancingStrokeShader;
+let particleModel;
+
+function instancingCallback() {
+	// a pseudorandom function from:

```

Comment:
Looks like we've got some mixed tabs + spaces here, could we change the tabs to spaces for consistency?